lib/ts/framework/h3/framework.ts

```diff
@@ -0,0 +1,201 @@
+        return path;
+    };
+}
+
+export class H3Response extends BaseResponse {
+    private event: H3Event;
+    private statusCode: number;
+
+    constructor(event: H3Event) {
+        super();
+        this.original = event;
+        this.event = event;
+        this.statusCode = 200;
+    }
+
+    sendHTMLResponse = (html: string) => {
+        if (this.event.node.res.writable) {
+            this.event.node.res.setHeader("Content-Type", "text/html");
+            this.event.node.res.statusCode = this.statusCode;
+            send(this.event, html);
+            // this.response.status(this.statusCode).send(Buffer.from(html));
+        }
+    };
+
+    setHeader = (key: string, value: string, allowDuplicateKey: boolean) => {
+        try {
+            let existingHeaders = this.event.node.res.getHeaders();
+            let existingValue = existingHeaders[key.toLowerCase()];
+
+            // we have the this.response.header for compatibility with nextJS
+            if (existingValue === undefined) {
+                this.event.node.res.setHeader(key, value);
+            } else if (allowDuplicateKey) {
+                this.event.node.res.setHeader(key, existingValue + ", " + value);
+            } else {
+                // we overwrite the current one with the new one
+                this.event.node.res.setHeader(key, value);
+            }
+        } catch (err) {
+            throw new Error("Error while setting header with key: " + key + " and value: " + value);
+        }
+    };
+
+    setCookie = (
+        key: string,
+        value: string,
+        domain: string | undefined,
+        secure: boolean,
+        httpOnly: boolean,
+        expires: number,
+        path: string,
+        sameSite: "strict" | "lax" | "none"
+    ) => {
+        setCookie(this.event, key, value, {
+            secure,
+            sameSite,
+            httpOnly,
+            expires: new Date(expires),
+            domain,
+            path,
+        });
+    };
+
+    /**
+     * @param {number} statusCode
+     */
+    setStatusCode = (statusCode: number) => {
+        if (this.event.node.res.writable) {
+            this.statusCode = statusCode;
+        }
+    };
+
+    sendJSONResponse = (content: any) => {
+        if (this.event.node.res.writable) {
+            this.event.node.res.setHeader("Content-Type", "application/json");
+            this.event.node.res.statusCode = this.statusCode;
+            send(this.event, content);
+        }
+    };
+}
+
+export interface SessionEvent extends H3Event {
+    context: {
+        session?: SessionContainerInterface;
+    };
+}
+
+export const middleware = eventHandler(async (event) => {
+    const supertokens = SuperTokens.getInstanceOrThrowError();
+    const request = new H3Request(event);
+    const response = new H3Response(event);
+    try {
+        await supertokens.middleware(request, response);
+    } catch (err) {
+        await supertokens.errorHandler(err, request, response);
```

Comment:
In the [discord logs](https://discord.com/channels/603466164219281420/995300294147711066/1064625720741085324) you had sent earlier, it seemed that the code execution came in the error handler (which is expected), and that this function wrote status code `401` to the `response` object, but you still got back a 404. This could possibly be because:
- Setting the status code on the `response` object and sending a reply doesn't actually send it to the client (maybe a bug in the `H3Response` implementation above)
- h3 routing may require you to define paths for all routes that can be called from the frontend, else it returns a `404`. This behaviour is similar to other frameworks like Hapi in which [we push all the routes dynamically as seen here](https://github.com/supertokens/supertokens-node/blob/master/lib/ts/framework/hapi/framework.ts#L232).